### PR TITLE
feat(env): default NOTEBOOKLM_STRICT_DECODE=1 + ADRs 011/012 (Tier 13 PR 13.9a)

### PR DESCRIPTION
**Pull request by teng-lin**
## Summary

Flips `NOTEBOOKLM_STRICT_DECODE` default from `"0"` to `"1"`. The `safe_index`-migrated batchexecute decoders now raise `UnknownRPCMethodError` (subclass of `RPCError`) on Google-side shape drift instead of degrading silently to `None`. Opt-out: `NOTEBOOKLM_STRICT_DECODE=0` for one release window.

Also lands two ADRs that this flip depends on:

- **ADR-011 — Schema validation policy.** Captures why fail-fast at the decoder boundary beats silent degradation for an undocumented-API client, and the retirement timeline for the opt-out.
- **ADR-012 — Implementation surface convention.** Pins the underscore-prefix policy (no-underscore = public, single-underscore = seam, double-underscore / `tests/` = test-only) with an explicit carve-out for `notebooklm.rpc/` and `notebooklm.cli/` (historical-internal subpackages with public-looking names whose status is documented in `docs/stability.md`).

This is the **parallel-safe portion** of the original t13-9 (per the 2026-05-19 phase plan amendment in `phase-2.md`). The deferred t13-9b will run after t13-8 and handles `__init__.py` / `__all__` audit / migration doc / `docs/python-api.md` / `docs/examples/` once final post-rename module names are settled.

## Test plan

- [x] `uv run pre-commit run --all-files` — clean (ruff legacy + ruff format)
- [x] `uv run mypy src/notebooklm --ignore-missing-imports` — clean (137 source files, 0 issues)
- [x] `uv run pytest -n auto` — 5490 passed, 20 skipped
- [x] `uv run pytest tests/integration tests/e2e -m vcr -n auto` — 152 passed, 4 skipped
- [x] New `tests/unit/test_strict_decode_default.py` — pins the default-on contract and parametrized truthy/falsy env values
- [x] 13 existing tests updated to set `NOTEBOOKLM_STRICT_DECODE=0` where they pin the legacy soft-mode contract (per ADR-007)

## Disjointness with in-flight tier-13 waves

This PR is intentionally scoped to be a no-conflict parallel branch alongside in-flight waves t13-4 → t13-8. Files touched:

- `src/notebooklm/_env.py`, `src/notebooklm/rpc/_safe_index.py` (docstring refresh only)
- `docs/adr/0011-*.md` (new), `docs/adr/0012-*.md` (new), `docs/adr/README.md`, `docs/configuration.md`, `docs/stability.md`, `CHANGELOG.md`
- `tests/unit/test_strict_decode_default.py` (new) + 13 existing test files

Explicitly NOT touched (would conflict with merged or in-flight waves): `src/notebooklm/__init__.py`, `_core*.py`, `_capabilities.py`, feature modules, `_session.py`, `_kernel.py`, `_session_contracts.py`, `client.py`, `docs/migration-tier-12-to-13.md`, `docs/python-api.md`, `docs/examples/`, any pre-existing internal `__all__`.

## Polish history

- **Stage 1 (code-simplifier)**: audited — no edits needed.
- **Stage 2 review**:
  - **Gemini**: APPROVE (no Critical/Important).
  - **Codex**: REQUEST CHANGES → flagged 2 Important findings (ADR-012's categorical "no-underscore = public" rule contradicted `docs/stability.md` treatment of `rpc/`/`cli/`; ADR-011 + `docs/configuration.md` overstated the migration as covering *every* internal call site). Both addressed in this PR. Also fixed: stale `rpc/_safe_index.py` module docstring + 2 stale "soft mode (default)" test comments.
  - **Claude reviewer**: terminated mid-edge-case pass; no Critical surfaced before stop.
- **Stage 3 (ultrathink)**: skipped per orchestrator decision after Stage 2 fixes — verify gate green, Codex's blockers addressed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Breaking Changes**
  * Schema validation is strict by default: mismatched API responses now raise a decoding/RPC error instead of silently returning empty/None results. Set NOTEBOOKLM_STRICT_DECODE=0 to temporarily restore legacy warn-and-return behavior for one release window.

* **Documentation**
  * Added decision records for schema-validation policy and module surface conventions; updated configuration and stability docs to reflect the strict default and opt-out semantics.

* **Tests**
  * Tests updated to explicitly pin soft-mode where legacy warn-and-return behavior is required.

[Review Change Stack](https://app.coderabbit.ai/change-stack/teng-lin/notebooklm-py/pull/856?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by teng-lin** [comment]
@claude review

File: docs/adr/0012-implementation-surface-convention.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+   internal — use `auth.py` (the facade)."
+
+## Decision
+
+Every Python module in `src/notebooklm/` belongs to exactly one of
+three categories, signalled by its filename:
+
+### 1. Public surface (no underscore prefix)
+
+```
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply docs/adr/0012-implementation-surface-convention.md]: Addressed in commit 5d1f868 — tagged both file-tree fenced code blocks (ADR-012 lines 81 and 121) with `text` language identifier and aligned the falsy-values list across `src/notebooklm/_env.py:34`, `docs/configuration.md` (table row + detail section), and `docs/adr/0011-...` to match the test parametrize set (`0`/`false`/`False`/`no`/`off`/`""`) plus an explicit note that anything not in the truthy set `{"1", "true", "True"}` is treated as non-truthy. Pre-commit + `test_strict_decode_default.py` (12 passed) green.

File: docs/adr/0011-schema-validation-policy.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+   already covers the strict-mode raise — the new default does not
+   force downstream callers to add a new except-clause unless they are
+   intentionally treating drift as a non-error sentinel.
+
+## Decision
+
+`_env.is_strict_decode_enabled()` defaults to `True` when
+`NOTEBOOKLM_STRICT_DECODE` is unset. Soft mode is reachable only via
+explicit `NOTEBOOKLM_STRICT_DECODE=0` (or any other non-truthy value:
+`"false"`, `"no"`, `""`).
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply docs/adr/0011-schema-validation-policy.md]: Addressed in commit 5d1f868 — tagged both file-tree fenced code blocks (ADR-012 lines 81 and 121) with `text` language identifier and aligned the falsy-values list across `src/notebooklm/_env.py:34`, `docs/configuration.md` (table row + detail section), and `docs/adr/0011-...` to match the test parametrize set (`0`/`false`/`False`/`no`/`off`/`""`) plus an explicit note that anything not in the truthy set `{"1", "true", "True"}` is treated as non-truthy. Pre-commit + `test_strict_decode_default.py` (12 passed) green.

File: docs/configuration.md [hunk 3]
```diff
@@ -150,7 +150,7 @@ be audited from one location.
...
 | `NOTEBOOKLM_QUIET_DEPRECATIONS` | Suppress stderr deprecation notices for deprecated CLI flags (e.g. `source add --mime-type` on file sources). Library-level `DeprecationWarning`s are unaffected. | Set to `1` to suppress; any other value (or unset) leaves the notice enabled. | individual CLI commands; see `NOTEBOOKLM_QUIET_DEPRECATIONS` section below |
-| `NOTEBOOKLM_STRICT_DECODE` | Toggle the decoder's drift behavior — warn-and-fallback (`0`, default) vs raise `UnknownRPCMethodError` (`1`/`true`/`True`). | Process env on each decode call. | `_env.is_strict_decode_enabled` |
+| `NOTEBOOKLM_STRICT_DECODE` | Toggle the decoder's drift behavior — raise `UnknownRPCMethodError` (unset/`1`/`true`/`True`, default since PR 13.9a) vs warn-and-fallback opt-out (`0`/`false`/`no`/`""`). | Process env on each decode call. | `_env.is_strict_decode_enabled` |
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply docs/configuration.md]: Addressed in commit 5d1f868 — tagged both file-tree fenced code blocks (ADR-012 lines 81 and 121) with `text` language identifier and aligned the falsy-values list across `src/notebooklm/_env.py:34`, `docs/configuration.md` (table row + detail section), and `docs/adr/0011-...` to match the test parametrize set (`0`/`false`/`False`/`no`/`off`/`""`) plus an explicit note that anything not in the truthy set `{"1", "true", "True"}` is treated as non-truthy. Pre-commit + `test_strict_decode_default.py` (12 passed) green.

File: docs/configuration.md [hunk 4]
```diff
@@ -320,18 +323,30 @@ the value at a step isn't indexable), behavior depends on
...
 |-------|----------|
-| `0` (default) | Log a warning with the failing path, `method_id`, `source` label, and a truncated repr of the data. Return `None` so legacy callers keep working. |
-| `1` / `true` / `True` | Raise `UnknownRPCMethodError` (a subclass of `DecodingError` / `RPCError`) with structured `method_id`, `path`, `source`, and `data_at_failure` attributes. |
-
-The default of `0` is a soft-rollout safeguard for this release while
-call sites migrate to defensive indexing. A future release will flip the
-default to `1` — set `NOTEBOOKLM_STRICT_DECODE=1` in your CI/staging
-environments now to catch drift early.
+| unset / `1` / `true` / `True` (default since PR 13.9a) | Raise `UnknownRPCMethodError` (a subclass of `DecodingError` / `RPCError`) with structured `method_id`, `path`, `source`, and `data_at_failure` attributes. |
+| `0` / `false` / `no` / `""` (opt-out) | Log a warning with the failing path, `method_id`, `source` label, and a truncated repr of the data. Return `None` so legacy callers keep working. |
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply docs/configuration.md]: Addressed in commit 5d1f868 — tagged both file-tree fenced code blocks (ADR-012 lines 81 and 121) with `text` language identifier and aligned the falsy-values list across `src/notebooklm/_env.py:34`, `docs/configuration.md` (table row + detail section), and `docs/adr/0011-...` to match the test parametrize set (`0`/`false`/`False`/`no`/`off`/`""`) plus an explicit note that anything not in the truthy set `{"1", "true", "True"}` is treated as non-truthy. Pre-commit + `test_strict_decode_default.py` (12 passed) green.

File: src/notebooklm/_env.py [hunk 5]
```diff
@@ -20,14 +20,22 @@
...
+    Strict mode is the default since PR 13.9a: when
+    ``NOTEBOOKLM_STRICT_DECODE`` is unset (or set to one of ``"1"``,
+    ``"true"``, ``"True"``), schema-drift helpers — notably
+    :func:`notebooklm.rpc.safe_index` — raise
+    :class:`~notebooklm.exceptions.UnknownRPCMethodError` on descent failure
+    so callers learn about Google-side shape drift immediately.
+
+    Set ``NOTEBOOKLM_STRICT_DECODE=0`` (or any other non-truthy value such
+    as ``"false"``, ``"no"``, or ``""``) to opt back into the legacy
+    warn-and-return-``None`` fallback for one release window. See
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/_env.py]: Addressed in commit 5d1f868 — tagged both file-tree fenced code blocks (ADR-012 lines 81 and 121) with `text` language identifier and aligned the falsy-values list across `src/notebooklm/_env.py:34`, `docs/configuration.md` (table row + detail section), and `docs/adr/0011-...` to match the test parametrize set (`0`/`false`/`False`/`no`/`off`/`""`) plus an explicit note that anything not in the truthy set `{"1", "true", "True"}` is treated as non-truthy. Pre-commit + `test_strict_decode_default.py` (12 passed) green.

**Comment by teng-lin** [comment]
@claude bot — addressed both nits in commit 03629a6:

1. **RST roles in ADR-011**: swapped `:func:notebooklm.rpc.safe_index` and `:class:~notebooklm.exceptions.UnknownRPCMethodError` to plain backtick code refs so they render as code in the GitHub UI instead of literal Sphinx role syntax.
2. **Redundant `setenv("0")` on `test_get_summary_happy_path_returns_string`**: removed; added a docstring note that the test is mode-agnostic (the happy-path shape never triggers `safe_index` drift) and points at the sibling `test_get_summary_happy_path_also_holds_under_strict` for the strict-mode no-regression contract.

Observation #3 (`test_safe_index.py` falsy set omits `"False"`) is deliberately left as-is: the new `tests/unit/test_strict_decode_default.py::test_falsy_values_disable_strict_mode` covers it, and the docstring at the top of the new file explains that it cross-pins (rather than duplicates) the sibling file's coverage.

pre-commit + targeted tests (test_get_summary_drift.py + test_strict_decode_default.py: 17 passed) green.

File: docs/configuration.md [hunk 6]
```diff
@@ -160,9 +160,12 @@ be audited from one location.
...
 
-**Boolean handling.** `NOTEBOOKLM_DEBUG_RPC` and `NOTEBOOKLM_STRICT_DECODE`
-treat `1` / `true` / `yes` (case-insensitive) as truthy; everything else is
-falsy. `NOTEBOOKLM_QUIET_DEPRECATIONS` requires the literal string `1`.
+**Boolean handling.** `NOTEBOOKLM_DEBUG_RPC` treats `1` / `true` / `yes`
+(case-insensitive) as truthy; everything else is falsy.
+`NOTEBOOKLM_STRICT_DECODE` treats `1` / `true` / `True` as truthy (the unset
+default also resolves to truthy post-PR 13.9a); everything else (`0`,
+`false`, `no`, `""`) is falsy and opts back into the soft-mode fallback.
+`NOTEBOOKLM_QUIET_DEPRECATIONS` requires the literal string `1`.
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply docs/configuration.md:169]: Addressed in commit 40ef816 — aligned the Boolean-handling paragraph at `docs/configuration.md:166` to list the same falsy set as the table row + detail section + ADR-011 (`0`/`false`/`False`/`no`/`off`/`""` + an explicit "any other non-truthy value" tail).
